inc/api.php

```diff
@@ -182,6 +183,39 @@ function stubEscapeFunctions()
             ]
         );
     }
+
+    /**
+     * Stub URL-related functions with default behavior.
+     */
+    function stubWpUrlFunctions($domain = 'example.org', $use_https = null)
+    {
+        $helper = new UrlsHelper($domain, $use_https);
+
+       stubs([
+            'home_url' => $helper->stubUrlCallback(),
+            'get_home_url' => $helper->stubUrlForSiteCallback(),
+            'site_url' => $helper->stubUrlCallback(),
+            'get_site_url' => $helper->stubUrlForSiteCallback(),
+            'admin_url' => $helper->stubUrlCallback('wp-admin', 'admin'),
+            'get_admin_url' => $helper->stubUrlForSiteCallback('wp-admin', 'admin'),
+            'content_url' => $helper->stubUrlCallback('wp-content', null, false),
+            'rest_url' => $helper->stubUrlCallback('wp-json'),
+            'get_rest_url' => $helper->stubUrlForSiteCallback('wp-json'),
+            'includes_url' => $helper->stubUrlCallback('wp-includes'),
+            'network_home_url' => $helper->stubUrlCallback(),
+            'network_site_url' => $helper->stubUrlCallback(),
+            'network_admin_url' => $helper->stubUrlCallback('wp-admin/network', 'admin'),
+            'user_admin_url' => $helper->stubUrlCallback('wp-admin/user', 'admin'),
+            'wp_login_url' => static function ($redirect = '', $force_reauth = false) use ($helper) {
```

Comment:
It looks like most of these don't take passed parameters to the original function into account, while WP does support passing parameters.

Stubbing these in the framework without taking the passes parameters into account will probably be confusing and lead to support overhead or to people just not using these stubs as they don't do what they expect.

Reply:
@jrfnl I think you need to look at it again :)

`$helper->stubUrlCallback()` and `$helper->stubUrlForSiteCallback()` return _callbacks_ that take respectively 2 (`$path`, `$schema`) and 3 parameters (`$blog_id`, `$path`, `$schema`), that are **the same core functions take**.

There are 2 exceptions: `content_url()` and `wp_login_url()`.

`wp_login_url()` takes still 2 arguments, but different from the others, and that is why it is stubbed differently.

`content_url()` takes 1 parameter: `$path`. That is why `stubUrlCallback()` accepts a 3rd parameter `$use_schema_arg` that defaults to `true`, but it is passed as `false` for `content_url()`.

To be noted:

- the `$blog_id` parameter for functions that accepts it is just ignored in stubbed functions because the base URL is taken from what is passed to the `stubWpUrlFunctions()` API function.
- When `$schema` param is something like `'admin'` or `'login'` it is taken into account only if HTTPs is not forced (passing `true` as `stubWpUrlFunctions()`'s 2nd parameter) and `force_ssl_admin()` is available (likely because stubbed separately).
- `rest_url()` and `get_rest_url()`  use `'rest'` as scheme. That is not relevant for the generation of the URL (in WP just like in the stubbed version), it is used in WP only as the 3rd parameter, `$orig_scheme`, passed to the `'set_url_scheme'` hook, something that I think is pretty safe to ignore for the stubbed version of the function.

**TL;DR**:  all the stubbed functions accept the same parameters and work very similarly to the WP counterparts. For some edge cases, it might be needed to stub `is_ssl()` and/or `force_ssl_admin()` separately.